gensim/matutils.py

```diff
@@ -382,6 +383,110 @@ def cossim(vec1, vec2):
     return result
 
 
+def isbow(vec):
+    """
+    Checks if vector passed is in bag of words representation or not.
+    """
+    if scipy.sparse.issparse(vec):
+        vec = vec.todense().tolist()
+    try:
+        id_, val_ = vec[0]
+        id_, val_ = int(id_), float(val_)
+    except Exception:
+        return False
+    return True
+
+
+def kullback_leibler(vec1, vec2, num_features=None):
+    """
+    A similarity metric between two probability distributions.
+    Returns a similarity in range <0,1> where values closer to 0 mean a higher similarity.
+    Uses the scipy.stats.entropy method to identify kullback_leibler convergence value.
+    If the distribution draws from a certain number of docs, that value must be passed.
+    """
+    if scipy.sparse.issparse(vec1):
+        vec1 = numpy.asarray(vec1.todense())
+    if scipy.sparse.issparse(vec2):
+        vec2 = numpy.asarray(vec2.todense())
+    if isinstance(vec1, numpy.ndarray):
+        vec1 = vec1.tolist()
```

Comment:
I wrote this bit a while ago and I just can't seem to remember why - I've removed it and tried and it's working fine, though I suspect I converted to lists then to catch a corner case somewhere. 
I'm sorry, I should've looked into that before asking you to review, will address in next commit.